### PR TITLE
chore(changelog): Add script to generate release notes

### PR DESCRIPTION
**Pull request by mttrbrts**
This PR adds documentation and a script for maintainers to use to automatically generate release note content from GitHub Pull Requests.

## Flags
- Note that the mapping in `package.json` uses the lowercase version of the GitHub Labels with unicode emoji characters rather than their short-codes.

Here is a sample output for the latest unreleased changes.

---

#### :sparkles: Enhancement
* `ui-contract-editor`
  * [#277](https://github.com/accordproject/web-components/pull/277) style(ui-contract-editor): Drag-Drop Handle Resize - I137 ([@K-Kumar-01](https://github.com/K-Kumar-01))
  * [#250](https://github.com/accordproject/web-components/pull/250) feat(ui-contract-editor):drag&drop disabled for embeded clauses ([@sanketshevkar](https://github.com/sanketshevkar))
* `storybook`, `ui-components`, `ui-concerto`, `ui-contract-editor`, `ui-markdown-editor`
  * [#247](https://github.com/accordproject/web-components/pull/247) feat(package): add repository information to package.json ([@Aniruddha-Shriwant](https://github.com/Aniruddha-Shriwant))
* `ui-markdown-editor`
  * [#237](https://github.com/accordproject/web-components/pull/237) feat(ui-markdown-editor): center align h1 tags - I146 ([@Udbhavbisarya23](https://github.com/Udbhavbisarya23))
  * [#225](https://github.com/accordproject/web-components/pull/225) style(ui-markdown-editor): remove link modal fade-in effect - I218 ([@K-Kumar-01](https://github.com/K-Kumar-01))
* `storybook`, `ui-markdown-editor`
  * [#229](https://github.com/accordproject/web-components/pull/229) style(ui-markdown-editor): resolve toolbar icon shrinkage - I228 ([@K-Kumar-01](https://github.com/K-Kumar-01))

#### :bug: Bug
* `ui-markdown-editor`
  * [#283](https://github.com/accordproject/web-components/pull/283) fix(ui-markdown-editor): styleblock refresh - #276 ([@d-e-v-esh](https://github.com/d-e-v-esh))
  * [#268](https://github.com/accordproject/web-components/pull/268) bug(ui-contract-editor): remove z-index of toolbar - I87 ([@Udbhavbisarya23](https://github.com/Udbhavbisarya23))
  * [#252](https://github.com/accordproject/web-components/pull/252) fix(ui-markdown-editor):multi-para to list proper indent - I158 ([@K-Kumar-01](https://github.com/K-Kumar-01))
  * [#258](https://github.com/accordproject/web-components/pull/258) fix(ui-markdown-editor): list formatting is preserved in headings ([@gatij10](https://github.com/gatij10))
  * [#222](https://github.com/accordproject/web-components/pull/222) fix(ui-markdown-editor): resolve hyperlink overflow - I219 ([@K-Kumar-01](https://github.com/K-Kumar-01))
* `ui-contract-editor`
  * [#251](https://github.com/accordproject/web-components/pull/251) fix(ui-contract-editor): Highlight read-only text clause template - i242 ([@K-Kumar-01](https://github.com/K-Kumar-01))
  * [#240](https://github.com/accordproject/web-components/pull/240) fix(ui-contract-editor) prevent sig variable edits on readOnly - I212 ([@sanketshevkar](https://github.com/sanketshevkar))
  * [#231](https://github.com/accordproject/web-components/pull/231) fix(ui-contract-editor): prevent clause drag and drop in list - I157 ([@sanketshevkar](https://github.com/sanketshevkar))
  * [#236](https://github.com/accordproject/web-components/pull/236) fix(ui-contract-editor): adjust tooltip z-index - I185 ([@K-Kumar-01](https://github.com/K-Kumar-01))
* `ui-components`
  * Long clause template names intersect with template logo ([@Michael-Grover](https://github.com/Michael-Grover))

#### :soap: Chore
* `storybook`, `ui-concerto`
  * [#273](https://github.com/accordproject/web-components/pull/273) feat(ui-concerto): Upgrade to Concerto v1.0 ([@mttrbrts](https://github.com/mttrbrts))

#### Committers: 8
- Aniruddha Shriwant ([@Aniruddha-Shriwant](https://github.com/Aniruddha-Shriwant))
- Gatij Taranekar ([@gatij10](https://github.com/gatij10))
- Kushal Kumar ([@K-Kumar-01](https://github.com/K-Kumar-01))
- Matt Roberts ([@mttrbrts](https://github.com/mttrbrts))
- Michael Grover ([@Michael-Grover](https://github.com/Michael-Grover))
- Sanket Shevkar ([@sanketshevkar](https://github.com/sanketshevkar))
- [@Udbhavbisarya23](https://github.com/Udbhavbisarya23)
- [@d-e-v-esh](https://github.com/d-e-v-esh)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,10 +44,20 @@
...
     "type": "git",
     "url": "https://github.com/accordproject/web-components"
+  },
+  "changelog": {
+    "labels": {
+      "type: breaking change 💥": ":boom: Breaking Change",
+      "type: enhancement ✨": ":sparkles: Enhancement",
+      "type: bug 🐛": ":bug: Bug",
+      "type: chore 🧼": ":soap: Chore",
+      "type: documentation 📝": ":memo: Documentation"
```
Comment (irmerk) [inline package.json:60]: So it looks like we should maybe do a more comprehensive mapping of [our labels](https://github.com/accordproject/web-components/labels) to match what is in [our current releases](https://github.com/accordproject/techdocs/wiki/Releases#release-notes-structure) (that related to `web-components`)?

File: DEVELOPERS.md [hunk 2]
```diff
@@ -47,4 +47,28 @@ npm run build
...
+
+1. You will need a GitHub [Personal Access Token](https://github.com/settings/tokens) with `public_repo` permissions.
+2. Review [all merged PRs since the last release](https://github.com/accordproject/web-components/pulls?q=is%3Apr+is%3Aclosed+is%3Amerged) to ensure that they are appropriately labelled with one of the following labels.
+    - `Type: Breaking Change 💥`
+    - `Type: Enhancement ✨`
+    - `Type: Bug 🐛`
+    - `Type: Chore 🧼`
+    - `Type: Documentation 📝`
+    
+> Note that `Type: Styling` and `Type: Feature Request` are not used, you should use `Type: Enhancement` instead.
```
Comment (irmerk) [inline DEVELOPERS.md:68]: Relatedly, why not use those types?

File: DEVELOPERS.md [hunk 3]
```diff
@@ -47,4 +47,28 @@ npm run build
...
+
+3. Run the following command from the root folder of an up-to-date local clone of this repositoryto generate the markdown content.
```
Comment (irmerk) [inline DEVELOPERS.md:70]: `repositoryto` -> `repository to`

File: DEVELOPERS.md [hunk 4]
```diff
@@ -47,4 +47,28 @@ npm run build
...
+```bash
+GITHUB_AUTH=[YOUR_PERSONAL_ACCESS_TOKEN] npm run changelog:unreleased
+```
+4. Copy the markdown output to the [GitHub release editor](https://github.com/accordproject/web-components/releases/new).
```
Comment (irmerk) [inline DEVELOPERS.md:74]: Does this output into the console or into a file?

**Review by irmerk — COMMENTED** [review]
Some questions...

File: package.json (hunk 1 above)
Comment (mttrbrts) [reply package.json:60]: I found that in cases where multiple labels have been applied, this tool adds the PR in multiple places. 

Personally, I also find the difference between a "Feature Request", "Enhancement" and "Styling" to be too subtle for a release note.

Also, "Feature Request" specifically doesn't make much sense as a category for a release note.

File: DEVELOPERS.md (hunk 2 above)
Comment (mttrbrts) [reply DEVELOPERS.md:68]: See reply above

File: DEVELOPERS.md (hunk 4 above)
Comment (mttrbrts) [reply DEVELOPERS.md:74]: Into the terminal, I'll clarify. Thanks

File: package.json (hunk 1 above)
Comment (mttrbrts) [reply package.json:60]: BTW, I also tried mapping "Feature Request" and "Styling" labels to ":sparkles: Enhancement" but that leads to a bug in the tool where we end up with 3 copies of the "Enhancement" section and the "Feature request" and "Styling" PRs are lost.

**Review by DianaLease — COMMENTED** [review]
This is awesome!

File: package.json (hunk 1 above)
Comment (irmerk) [reply package.json:60]: Well I think `Feature Request` would be `New Feature`.

So actually, what happens to merged PRs that are `Type: Feature Request` right now? Are they not included?

Comment (mttrbrts) [reply package.json:60]: > So actually, what happens to merged PRs that are `Type: Feature Request 🛍️` right now? Are they not included?

That's right, they get ignored.

What's the difference between `Type: Enhancement ✨ `, `Type: Feature Request 🛍️`  (as a GitHub label ) and `🚀 New Feature` ?

Comment (irmerk) [reply package.json:60]: `Type: Enhancement ✨` and `Type: Feature Request 🛍️`` are labels for PRs and Issues, `🚀  New Feature` is a tag in the release note. More PRs are tagged `Type: Feature Request 🛍️`` than `Type: Enhancement ✨`, so we don't want to be ignoring those.

Comment (mttrbrts) [reply package.json:60]: Let me rephrase...

In a release note, personally, I wouldn't expect a category called "Feature Request". By the time that it gets to that stage, the change is a feature (or an enhancement). I agree that "Feature Request" makes sense as a label for issues, but less so for PRs.

In that context, we shouldn't retain both and map them to the same `🚀 New Feature` section in a release note because of the bug that I mentioned in `lerna-changelog`. We can only have unique targets in the mapping.

So my proposal is to not use the label `Type: Feature Request 🛍️` for PRs and instead use `Type: Enhancement ✨`. An alternative would be to also label every PR that has `Type: Feature Request 🛍️` with `Type: Enhancement ✨` (because the former would be ignored)

Comment (irmerk) [reply package.json:60]: Okay yeah this makes sense. So Issues => `Type: Feature Request 🛍️`, PRs => `Type: Enhancement ✨`.

Could you as a next step update https://github.com/accordproject/web-components/wiki to have the `Release Workflow` Guide to not point to [`techdocs`](https://github.com/accordproject/techdocs/wiki/Releases) and rather fork that information and update it with what this PR changes? We'll want to have something definitive for maintainers to reference when both creating a release note and also labeling PRs.

**Review by irmerk — APPROVED** [review]
